kyaml/setters2/set.go

```diff
@@ -295,7 +295,7 @@ func validateAgainstSchema(ext *CliExtension, sch *spec.Schema) error {
 		var format string
 		// Only add quotes around the value is string is one of the
 		// types in the schema.
-		if sch.Type.Contains("string") {
+		if sch.Type.Contains("string") || ext.Setter.Value[len(ext.Setter.Value)-1] == ':' {
```

Comment:
please move this logic to a method with a descriptive name
```
if shouldQuoteSetterValue(..)
```
also, i didn't look at the above code, but this will fail if there's a chance ext.Setter.Value has len zero.

Reply:
+1 for new method and null check. And the reason for this is yaml parser is getting confused by : at the end. 

You can use strings.HasSuffix() which is cleaner. 

You should also check for setting list values.